### PR TITLE
observability: centralize docs on cloud monitoring, add loki

### LINKED ISSUE #26787: Create handbook entry for Loki
We are using Grafana Cloud and Loki for logging. This integration may have been a POC at one point but multiple teams are now using it (search team(s) & dev-experience). 

We should document what it does and how to interact with it 
- How to request access (and who should have access) 
- How to purge data 
- Brief overview of integration points ( how it ingests data ) 

Importantly, we should update this page https://handbook.sourcegraph.com/bizops/customer_data_policy if this might ingest customer data ( I believe it does)

### PR DESCRIPTION
**Pull request by bobheadxi**
- rename `observability/cloud_monitoring` to `observability/cloud` for consolidation
- add notes on `/-/debug/grafana`, Grafana Cloud, Loki for cloud, Loki for CI
- migrate general dashboard docs to `docs.sourcegraph.com`:  https://github.com/sourcegraph/sourcegraph/pull/28692
- updates notes to use admonition
- rename `sourcegraph.com` instance to `Sourcegraph Cloud`

Closes https://github.com/sourcegraph/sourcegraph/issues/26787

Preview: https://deploy-preview-1284--sourcegraph-handbook.netlify.app/product-engineering/engineering/observability/cloud

### CODE REVIEW

File: content/product-engineering/engineering/observability/cloud.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Logs in Grafana Cloud is provided by [Grafana Loki](https://grafana.com/oss/loki/), a logs aggregation system that uses a PromQL-like query language called [LogQL](https://grafana.com/docs/loki/latest/logql/).
+
+Loki allows you to easily query for logs, filter for fields within structured logs, and even generate metrics from logs. The [official LogQL documentation](https://grafana.com/docs/loki/latest/logql/) provides a complete reference, or you can refer to [this cheatsheet](https://megamorf.gitlab.io/cheat-sheets/loki/) for a brief overview.
+
+#### Cloud logs
+
+The Loki instance in Grafana Cloud is currently configured to ingest logs from Sourcegraph Cloud pushed from [`grafana-agent`'s Loki configuration](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/configure/grafana-agent/grafana-agent.ConfigMap.yaml#L58). To query these, you can start with a LogQL query like:
+
+```logql
```
Comment (jhchabran) [inline content/product-engineering/engineering/observability/cloud.md:33]: That's neat to have added an example 👌

**Review by jhchabran — COMMENTED** [review]
LGTM :)

**Review by jhchabran — APPROVED** [review]

File: content/product-engineering/engineering/deployments/instances.md [hunk 2]
```diff
@@ -13,11 +13,11 @@ For deployments of Sourcegraph we manage for customers, see [managed instances](
 
 Also see [playbooks](./playbooks.md) for common actions related to operating our Sourcegraph deployments.
 
-## sourcegraph.com
+## Sourcegraph Cloud
 
 [![Build status](https://badge.buildkite.com/ef1289610fdd05b606bf1e57a034af2365c7b09c95ac6121f9.svg)](https://buildkite.com/sourcegraph/deploy-sourcegraph-dot-com)
 
-This deployment is also colloquially referred to as "Sourcegraph Cloud", "Cloud", and "dot-com". It is the public deployment available to the public at [sourcegraph.com/search](https://sourcegraph.com/search).
+This deployment is also colloquially referred to as "sourcegrpah.com", "Cloud", and "dot-com". It is the public deployment available to the public at [sourcegraph.com/search](https://sourcegraph.com/search).
```
Comment (daxmc99) [inline content/product-engineering/engineering/deployments/instances.md:20]: Sourcegraph.com 😅

**Review by daxmc99 — COMMENTED** [review]
LGTM!

File: content/product-engineering/engineering/deployments/instances.md (hunk 2 above)
Comment (bobheadxi) [reply content/product-engineering/engineering/deployments/instances.md:20]: OH NO

Comment (bobheadxi) [reply content/product-engineering/engineering/deployments/instances.md:20]: I mean **_I_** refer to it exclusively as `sourcegrpah.com`, but here's a fix anyway: https://github.com/sourcegraph/handbook/pull/1308

**Pull request merged**